bwl: dwpal: correct parsing of monitor STA disconnected event

arnout opened the pull request:
The parsing of the STA-Disconnected event in the monitor was not done
correctly.

Up to recently this was not a problem, since the event was anyway not
used in the monitor. However, it is needed now to stop the automatic
monitoring.

In addition, commit 700df9b59ffa1695beef1c02d9f28dc8141584e8 changed the
implementation of mac_from_string - it now raises an exception if the
string is not a valid MAC address instead of just returning ZERO_MAC.
Thus, every disconnected event aborts the monitor.

adam1985d commented on common/beerocks/bwl/dwpal/mon_wlan_hal_dwpal.cpp:
```diff
@@ -1147,10 +1147,11 @@ bool mon_wlan_hal_dwpal::process_dwpal_event(char *buffer, int bufLen, const std
             {NULL /*opCode*/, &numOfValidArgs[0], DWPAL_STR_PARAM, NULL, 0},
-            {(void *)MACAddress, &numOfValidArgs[1], DWPAL_STR_PARAM, NULL, sizeof(MACAddress)},
+            {(void *)VAP, &numOfValidArgs[1], DWPAL_STR_PARAM, NULL, sizeof(VAP)},
+            {(void *)MACAddress, &numOfValidArgs[2], DWPAL_STR_PARAM, NULL, sizeof(MACAddress)}
```
missing the ',' at the end of the field
this causes the build failure

tomereli commented on common/beerocks/bwl/dwpal/mon_wlan_hal_dwpal.cpp:
```diff
@@ -1160,6 +1161,7 @@ bool mon_wlan_hal_dwpal::process_dwpal_event(char *buffer, int bufLen, const std
             return false;
         }
 
+        LOG(DEBUG) << "vap_id     : " << VAP;
```
Perhaps this is the reason for the build failure?
```suggestion
        LOG(DEBUG) << "vap_id     : " <<std::string(VAP);
```

adam1985d replied:
no - I have commented on the cause of failure

arnout committed:
bwl: dwpal: correct parsing of monitor STA disconnected event

The parsing of the STA-Disconnected event in the monitor was not done
correctly.

Up to recently this was not a problem, since the event was anyway not
used in the monitor. However, it is needed now to stop the automatic
monitoring.

In addition, commit 700df9b59ffa1695beef1c02d9f28dc8141584e8 changed the
implementation of mac_from_string - it now raises an exception if the
string is not a valid MAC address instead of just returning ZERO_MAC.
Thus, every disconnected event aborts the monitor.

Signed-off-by: Arnout Vandecappelle (Essensium/Mind) <arnout@mind.be>
